### PR TITLE
SWC-6763

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

File: packages/synapse-react-client/src/components/error/SendMessageToEntityOwnerDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    if (!currentUserProfile) {
+      console.error(
+        'Attempted to send message without a user profile. This should never happen.',
+      )
+      return
+    }
+    const { firstName, lastName, userName } = currentUserProfile
+    sendMessage({
+      entityId,
+      subject: `${firstName} ${lastName} ${userName} Requested Access to ${entityId}`,
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/error/SendMessageToEntityOwnerDialog.tsx:58]:
```suggestion
      subject: `${firstName} ${lastName} (@${userName}) Requested Access to ${entityId}`,
```

File: packages/synapse-react-client/src/components/error/SendMessageToEntityOwnerDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      body: `Requesting access to https://www.synapse.org/Synapse:${entityId}\n\n${message}`,
+    })
+  }, [message, currentUserProfile, sendMessage])
+  return (
+    <Dialog maxWidth="md" open={isOpen} onClose={onHide} sx={{ zIndex: 1500 }}>
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/error/SendMessageToEntityOwnerDialog.tsx:63]: Is the zIndex necessary?

Consider just using `ConfirmationDialog` for style consistency with other dialogs in SRC

**Review by nickgros — APPROVED** [review]
LGTM, minor suggestions and questions

File: packages/synapse-react-client/src/synapse-client/SynapseClient.ts [hunk 3]
```diff
@@ -5356,12 +5385,12 @@ export async function getAnnotationColumnModels(
   return getAllOfNextPageTokenPaginatedService(fn)
 }
 
-export async function sendMessage(
+const getMessageToUser = async (
```
[the remark this answers is not in the document]
Comment (jay-hodgson) [reply packages/synapse-react-client/src/synapse-client/SynapseClient.ts:5388]: Renaming to createMessageToUserObject()

File: packages/synapse-react-client/src/components/error/SendMessageToEntityOwnerDialog.tsx (hunk 2 above)
Comment (jay-hodgson) [reply packages/synapse-react-client/src/components/error/SendMessageToEntityOwnerDialog.tsx:63]: Removed zIndex, and converted to be a ConfirmationDialog!

**Pull request merged**